src/symboltable.c

```diff
@@ -18,15 +18,15 @@ void insertVariable(char *var, int value, int type) {
         st = (Symbol *)malloc(sizeof(Symbol) * ST_SIZE);
     }
     Symbol s;
-    s.variable = (char *)malloc(sizeof(char) * strlen(var));
+    s.variable = (char *)malloc(sizeof(char) * (strlen(var) + 1));
     // s.value = (char *)malloc(sizeof(char) * MAX_TOK_LEN);
 
     strcpy(s.variable, var);
     s.value = value;
     s.type = type;
 
-    if(getSymbol(var) != NULL) {
-        int pos = getPos(var);
+    int pos = getPos(var);
```

Comment:
I think this is not necessary. `getSymbol` clearly states the intent and is easier to read.

Reply:
i think this is better and if you want i suggest to rename `getPos` to `getSymbolPos`, it will better better performance and more readable.

Reply:
It won't have any performance improvements as both codes are the same with O(n) complexity. But you can change the function to `getSymbolPos` for better readability.

Reply:
yes both function almost the same, but testing `getSymbol` the `getPos` make it search twice.
so what i did is searching once. and still readable